rust/kernel/types.rs

```diff
@@ -71,3 +74,64 @@ macro_rules! cstr {
         unsafe { $crate::CStr::new_unchecked(s) }
     }};
 }
+
+/// Used to convert an object into a raw pointer that represents it.
+///
+/// It can eventually be converted back into the object. This is used to store objects as pointers
+/// in kernel data structures, for example, an implementation of [`FileOperations`] in `struct
+/// file::private_data`.
+pub trait PointerWrapper<T> {
```

Comment:
The type parameter `T` is not needed, see the discussion between @wedsonaf  and me: https://github.com/Rust-for-Linux/linux/pull/109#discussion_r598878381